capstone_project/src/test/java/com/google/sps/SchoolTest.java

```diff
@@ -100,4 +144,40 @@ public void checkEqualsWorksForSet() {
     int expected = 3;
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void testUploadingSchoolsWithNoDuplicatesInDatastore() {
+    when(request.getParameter("name-input")).thenReturn("UCB");
+    when(request.getParameter("latitude-input")).thenReturn("37.871942");
+    when(request.getParameter("longitude-input")).thenReturn("-122.258476");
+
+    try {
+      mapServlet.doPost(request, response);
+    } catch (IOException ex) {
+      System.out.println(ex.toString());
+    }
```

Comment:
Make this method throw the IOException `public void testUploadingSchoolsWithNoDuplicatesInDatastore() throws IOException` that way, in case it were to throw an io exception, the test would fail because that's what should happen, a test should fail if there is an exception (like Null pointer). same for the one below